### ISSUE TITLE
Rule proposal: disallow chained variable assignment

### ISSUE THREAD

**Issue by lencioni** [eslint/eslint#6932]
> When the rules will warn. Include a description as well as sample code.

Chained variable assignment creates implicit global variables. Although these should be caught by the `no-undef` rule, people may want to prevent chained variable assignment for readability purposes.

```js
// Bad (this fails no-undef)
var a = b = c = 1;

// Bad (this passes no-undef)
var b;
var c; 
var a = b = c = 1;

// Good
var a = 1;
var b = a;
var c = a;
```

> Whether the rule prevents an error or is stylistic.

This rule prevents an error if the `no-undef` rule is not enabled, otherwise it is stylistic and similar to the `one-var` rule.

> Why the rule should be in the core instead of creating a custom rule.

This is a basic JavaScript syntax rule that is generally applicable to anyone writing JavaScript.

> Are you willing to create the rule yourself?

Sure.

**Comment by platinumazure** [comment]
This seems pretty easy to implement, just warning when the init of VariableDeclarator contains an AssignmentExpression. The only thing I'm not sure about is whether we have a rule that should be enhanced, instead of creating a new one. Labeled for now as a new rule request.

**Comment by ljharb** [comment]
I think it makes the most sense as a new rule. one-var might work, but even when you don't want to require or disallow multiple var statements, you may want to govern this pattern. no-undef also might work, but this is not saying "don't use undefined/global vars", it's saying "don't accidentally create some global vars with one local".

**Comment by platinumazure** [comment]
@ljharb What you're saying makes sense and is basically what I had been thinking in my head earlier. I just wanted to give the ESLint team a chance to jump in and let me know if I had missed something.

I'll probably champion this, once we figure out how exactly we want to do ti.

**Comment by mysticatea** [comment]
I think this is similar to #6424.

**Comment by lencioni** [comment]
@mysticatea ah yes! My searches did not discover that issue. I'll close this one in favor of that one.

**Comment by mysticatea** [comment]
Oops, #6424 is similar but not same.
This is about assignments inside of initializers.
#6424 is about assignments inside of assignments.
